worker/download.go

```diff
@@ -209,18 +209,35 @@ func (mgr *downloadManager) DownloadObject(ctx context.Context, w io.Writer, o o
 	// create the cipher writer
 	cw := o.Key.Decrypt(w, offset)
 
-	// create the trigger chan
+	// create response and trigger chan
 	nextSlabChan := make(chan struct{}, 1)
 	nextSlabChan <- struct{}{}
+	responseChan := make(chan *slabDownloadResponse)
+
+	// ensure channels are cleaned up properly
+	var wg sync.WaitGroup
+	ctx, cancel := context.WithCancel(ctx)
+	defer func() {
+		// make sure everything cancels if the download is done
+		cancel()
+		// wait for all goroutines to finish and cleanup resources
+		go func() {
```

Comment:
Do we need this goroutine? To me it seems like all the goroutines should stop pretty much immediately when we cancel the context. That way we will also notice if we accidentally introduce a bug where we leak goroutines.

Reply:
Hm, I actually added it later because I figured there was no point in blocking a finished download, even if this `wg.Wait` never takes long to complete. I'll remove it though